server/blob/blob-export-file/src/test/java/org/apache/james/blob/export/file/SimpleBlobExportMechanismTest.java

```diff
@@ -0,0 +1,132 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.export.file;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import javax.mail.Message;
+import javax.mail.internet.InternetAddress;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.api.ObjectStoreException;
+import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.filesystem.api.FileSystem;
+import org.apache.james.server.core.configuration.Configuration;
+import org.apache.james.server.core.filesystem.FileSystemImpl;
+import org.apache.james.util.MimeMessageUtil;
+import org.apache.mailet.base.MailAddressFixture;
+import org.apache.mailet.base.test.FakeMailContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SimpleBlobExportMechanismTest {
+    private static final byte[] BLOB_CONTENT = "blob_content".getBytes(StandardCharsets.UTF_8);
+
+    private BlobStore blobStore;
+    private FakeMailContext mailetContext;
+    private FileSystem fileSystem;
+    private SimpleBlobExportMechanism testee;
+
+    @BeforeEach
+    void setUp() {
+        mailetContext = FakeMailContext.builder().postmaster(MailAddressFixture.POSTMASTER_AT_JAMES).build();
+        blobStore = new MemoryBlobStore(new HashBlobId.Factory());
+
+        Configuration configuration = Configuration.builder()
+            .workingDirectory("../")
+            .configurationFromClasspath()
+            .build();
+        fileSystem = new FileSystemImpl(configuration.directories());
```

Comment:
Are you cleaning temporary files?

Reply:
I think FileSystem(Impl) provides no clean way to do this.

This seems to be a recurring issue in tests relying on this class.

Do you have suggestions on how to do this?

Reply:
I did rework what you contributed to make this lovely extension rely on the real fileSystemImpl and not create yet-another fileSystem implementation.

Also an extension should resolve an API and not an implementation of it: this makes the test more generic.